docs/references/architecture/adr-113-mempool-lanes.md

```diff
@@ -3,43 +3,134 @@
+
+Note that improving the dissemination protocol to reduce bandwidth and/or latency is a separate
+concern and falls outside the scope of this proposal.
+
+## Properties
+
+Before jumping into the design of the proposal, we define more formally the properties supported by
+the current implementation of the mempool. Then we state what properties the new mempool should
+offer to guarantee the desired QoS.
+
+The following definition is common to all properties. 
+
+:memo: _Definition_: Given any two different transactions `tx1` and `tx2`, we say that `tx1` is
+*processed and disseminated before* `tx2`, when:
+- `tx1` is reaped from the mempool to form a block proposal before `tx2`,
+- `tx1` is rechecked before `tx2`, and
+- `tx1` is disseminated to a peer before `tx2`.
+
+Note that in the current implementation there is one dissemination routine per peer, so it could
+happen that `tx2` is sent to a peer before `tx1` is sent to a different peer.
+
+### Current mempool
+
+As stated above, the current mempool offers a best-effort FIFO ordering of transactions. We state
+this property as follows.
+
+:parking: _Property_ **FIFO ordering of transactions**: We say that the mempool makes a best effort
+in maintaining the FIFO ordering of transactions when transactions are validated, processed, and
+disseminated in the same order in which the mempool has received them.
+
+More formally, given any two different transactions `tx1` and `tx2`, if the mempool receives `tx1`
+before receiving `tx2`, then:
+- `tx1` will be validated against the application (via `CheckTx`) before `tx2`, and
+- `tx1` will be processed and disseminated before `tx2`.
+
+Note that a node's mempool can receive a transaction either from a `broadcast_tx_*` RPC endpoint or
+from a peer.
+
+### Mempool with QoS
+
+With the introduction of *transaction classes*, the main goal is to guarantee that certain
+transactions have lower latency than others. 
+
+:memo: _Definition_: a *transaction class* is a disjoint set of transactions having some common
+characteristics as defined by the application.
+
+A transaction may only have one class. If it is not assigned any specific class, it will be assigned
+a *default class*, which is a special class always present in any set of classes. Because no
+transaction can belong to two or more classes, transaction classes form a disjoint set. Also, all
+transactions in the mempool are the union of the transactions in all classes.
+
+:memo: _Definition_: Each class has a *priority* and two classes cannot have the same priority. Then
+all classes can be ordered by priority.
 
-Try using "QoS" term. Because that's what it actually is.
+Now, given these definitions, we want the proposed QoS mechanism to offer the following property:
 
-We need problem definition.
+:parking: _Property_ **Priorities between classes**: Transactions belonging to a certain class will
+be processed and disseminated before transactions belonging to another class with lower priority.
```

Comment:
> processed and disseminated

and included in a block?

Reply:
tx1 being included in a block before tx2 is a consequence of this property. It is mentioned two paragraphs below.